backend/api/Services/DocumentService.cs

```diff
@@ -128,6 +129,31 @@ public async Task<DocumentUploadResponse> UploadActivityDocumentAsync(long activ
             return response;
         }
 
+        public async Task<bool> UpdateActivityDocumentMetadataAsync(long documentId, DocumentUpdateMetadataRequest updateRequest)
+        {
+            this.Logger.LogInformation("Updating document for single activity");
+            this.User.ThrowIfNotAuthorized(Permissions.DocumentEdit);
+
+            // update metadata of document
+            IList<Task<ExternalResult<DocumentMetadata>>> metadataTasks = new List<Task<ExternalResult<DocumentMetadata>>>();
+            foreach (var metadata in updateRequest.DocumentMetadata)
```

Comment:
does this work when there is no metadata associated to a document type (I had some issues on my local where similar logic on the create side was throwing an NPE.

Reply:
added check